sdv/constraints/tabular.py

```diff
@@ -211,74 +214,96 @@ class GreaterThan(Constraint):
         drop (str):
             Which column to drop during transformation. Can be ``'high'``,
             ``'low'`` or ``None``.
-        high_is_scalar(bool or None):
-            Whether or not the value for high is a scalar or a column name.
-            If ``None``, this will be determined during the ``fit`` method
-            by checking if the value provided is a column name.
-        low_is_scalar(bool or None):
-            Whether or not the value for low is a scalar or a column name.
-            If ``None``, this will be determined during the ``fit`` method
-            by checking if the value provided is a column name.
+        scalar (str):
+            Which value is a scalar. Can be ``'high'``, ``'low'`` or ``None``.
+            If ``None`` then both ``high`` and ``low`` are column names.
     """
 
-    _diff_column = None
+    _diff_columns = None
     _is_datetime = None
-    _column_to_reconstruct = None
+    _columns_to_reconstruct = None
 
     def __init__(self, low, high, strict=False, handling_strategy='transform',
-                 fit_columns_model=True, drop=None, high_is_scalar=None,
-                 low_is_scalar=None):
+                 fit_columns_model=True, drop=None, scalar=None):
+
+        if isinstance(low, list) and isinstance(high, list):
```

Comment:
I think this validations block is still too complex.

I would suggest these changes:

1. Move all the validations to a dedicated classmethod method.

    ```python3
    self.low, self.high, self.constraint_columns = self._validate_inputs(low, high, scalar)
    ```

    Another option is to make it an instance method that updates the arguments internally:

    ```python3
    self._validate_inputs(low, high, scalar)   # this sets `self.low` and the others internally
    ```

2. I would validate the `scalar=None` explicitly, and raise an exception if `scalar` is none of the expected values.

3. I would implement the asymmetric validations as another, generic static method.

    This method would expect a scalar value and column names, but without knowing which
    one is high and which one is low, and just validate their types, and would be called with
    `high` and `low` put in the right place depending on the direction. The `scalar` value can
    still be passed as a string to write the right error messages.

Altogether it would be something like:

```python3
@staticmethod
def _validate_scalar(scalar_column, column_names, scalar):
    """Validate scalar comparison inputs.

    - Make sure that the scalar column is not a list and raise the proper error if it is.
    - If the `column_names` is not a list it would make it a list.
    - Return both the scalar column and column names with the right format
    """
    ...

@classmethod
def _validate_inputs(cls, low, high, scalar):
    if scalar is None:
        # validate that neither high nor low are lists and 
    if scalar == 'low':
        low, high = cls._validate_scalar(scalar_column=low, column_names=high, scalar=scalar)
        constraint_columns = tuple(high)
    elif scalar == 'high':
        high, low = cls._validate_scalar(scalar_column=high, column_names=low, scalar=scalar)
        constraint_columns = tuple(low)
    else:
        raise ValueError(...)

    return low, high, constraint_columns

def __init__(self, ...):
    self.low, self.high, self.constraint_columns = self._validate_inputs(low, high, scalar)
```

Reply:
I like the idea of moving the validation to a class method `_validate_inputs`, and I think the class method should return the validated inputs (the first option), as the name suggests. If we want to update the arguments internally in this method, we should make the name something like `_validate_inputs_and_initialize`, but I think the first way is more explicit.